docs/interactions.rst

```diff
@@ -0,0 +1,274 @@
+
+.. code-block:: python
+
+  from pincer import Client
+  from pincer.commands import command
+
+  class Bot(Client):
+    @command
+    async def some_command(self):
+      pass
+
+  bot = Bot("TOKEN")
+  bot.run()
+
+This registers a command called "some_command". Its pretty useless right now so lets take a closer look at what else you can do.
+
+.. note::
+    If you aren't seing your command register, its likely because it takes up to one hour for a global slash command.
+    Specify the guild in the command decorator to have application commands register instantly.
+
+To register your command a guild do:
+
+.. code-block:: python
+
+    @command(guild=SOME_NUMBER)
+    async def some_command(self):
+        pass
+
+Sending messages to the user is extemely easy. What you want to return is inferred by the object's return type. A :class:`str` can be returned to send a text message.
+Heres a simple ping command.
+
+.. code-block:: python
+
+    @command
+    async def ping(self):
+        return "pong"
+
+
+If you need access to more information, you can pass in the :data:`ctx` object.
+
+.. note::
+    `ctx` and `self` should be those exact names or the correct value will not be passed in.
+
+.. code-block:: python
+
+    from pincer.objects import MessageContext
+    ...
+
+    @command
+    async def hello(self, ctx: MessageContext):
+        # Returns the name of the user that initiated the interaction
+        return f"hello {ctx.author}"
+
+Application Command Types
+-------------------------
+Pincer provides an API for all three interaction command types. The only thing that varies is the function signiture.
+
+.. code-block:: python
+
+    from pincer.commands import command, user_command, message_command
+    from pincer.objects import MessageContext, UserMessage, User
+    ...
+
+    @command
+    # Can have any amount of inputs
+    async def ping(self, ctx: MessageContext, arg1: str, arg2: str):
+        return "pong"
+
+    # Must have a parameter for users. User can be a GuildMember. All User
+    # methods are available to GuildMember because GuildMember inherits from
+    # User.
+    async def user_ping(self, ctx: MessageContext, user: User):
+        return "pong"
+
+    # Must have a parameter for messages.
+    async def message_ping(self, ctx: MessageContext, message: UserMessage):
+        return "pong"
+
+Interaction Timeout
+-------------------
+Interactions time out after 3 seconds. To extend the timeout to 15 minutes you can run :meth:`ack` from
+:class:`~pincer.objects.MessageContext`. :class:`~pincer.objects.app.interaction_flags.InteractionFlags` is available in this method.
+
+Arguments
+---------
+
+Arguments are more variables in the command. Notice how `word` is typehinted as string.
+Pincer uses type hints to infer the argument type that you want.
+
+.. code-block:: python
+
+    @command
+    async def say(self, word: str):
```

Comment:
I don't think this function needs the return to be typehinted because the return value goes into a magic void where it's never looked at. I think added `-> str` will also make it a little more confusing on what I mean when I say the type is inferred.

Reply:
I agree with Lunar here. 
Also it would just be more confusing for a beginner. We have to think that many people who are using this are using Python for the first time or at least beginners, because they want to make a discord bot. That's how some people started Python by creating a bot with discord.py. We have an obligation to maintain that ease and continue the legacy. Of course there are many people who know Python well and are using Pincer but we need to keep in mind that it should be easy to understand.